File: C/Codeforces/191107_DIV2/!C.cpp
#include <iostream>
#include <cstdio>
#include <string>

using namespace std;

long long n;
int mem[1000000] = {0};

long long solution(){
    long long max_num = 0;
    long long tmp;
    long long cnt = 1;
    for(long long i=2;i*i<=n;i++){
        if (n % i == 0 && mem[i] == 0){
            cnt++;
            for (long long j=1;j<=n/i;j++){
                mem[i*j] = cnt;
            }
        }        
    }

    if (cnt == 0){
        cnt = n;
    }
    
    return cnt;
}


int main(){
    freopen("input.txt","r",stdin);

    cin >> n;
    cout << solution();

    return 0;
}

File: Codeforces/#680_DIV2/!C.py
import time

ptime = time.time_ns()

import sys
sys.stdin = open('input.txt','r')


#####################################################


# p = ax
# x = bq + r (r != 0)
# p = a(bq + r)
#   = abq + ar







def solution(p, q):
    max_num = -1

    if p % q != 0:
        return p

    x = q
    while x > 1:
        for div in range(2,q+1):
            if x % div == 0:
                x = x // div
                if p % x == 0:
                    return x
                break

    return -1

t = int(input())
for tt in range(t):
    p, q = map(int, input().split())
    print(solution(p, q))


#####################################################

print('time',time.time_ns()-ptime)

